llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -4075,24 +4367,80 @@ bool MemProfContextDisambiguation::applyImport(Module &M) {
                      << ore::NV("Attribute", AllocTypeString));
           }
         } else if (!CallsiteContext.empty()) {
-          // Consult the next callsite node.
-          assert(SI != FS->callsites().end());
-          auto &StackNode = *(SI++);
-
+          if (!CalledFunction) {
```

Comment:
done on moving this handling to a helper. I made ICallAnalysis a private member, however it is initialized by another helper at the beginning of applyImport, since it isn't needed in other cases that use this class (e.g. fullLTO). I didn't move ICallAnalysisMap (now ICallAnalysisInfo) since that is a per-function data structure.